src/ucp/core/ucp_ep.c

```diff
@@ -1859,39 +1859,70 @@ int ucp_ep_config_lane_is_peer_match(const ucp_ep_config_key_t *key1,
 
 static ucp_lane_index_t
 ucp_ep_config_find_match_lane(const ucp_ep_config_key_t *key1,
-                              const ucp_rsc_index_t *dst_rsc_indices1,
                               ucp_lane_index_t lane1,
-                              const ucp_ep_config_key_t *key2,
-                              const ucp_rsc_index_t *dst_rsc_indices2)
+                              const ucp_ep_config_key_t *key2)
 {
     ucp_lane_index_t lane_idx;
 
     for (lane_idx = 0; lane_idx < key2->num_lanes; ++lane_idx) {
-        if (ucp_ep_config_lane_is_peer_match(key1, lane1, key2, lane_idx) &&
-            ucp_ep_lane_is_dst_index_match(dst_rsc_indices1[lane1],
```

Comment:
IMO we need to check tl_name_csum match, before returning a matching lane here. Otherwise we may call is_connected with different transports, which parse the UCT address differently

Reply:
as discussed, added assert for tl_name_csum